src/main/java/Sinep/Sinep.java

```diff
@@ -0,0 +1,322 @@
+    static String nl = System.lineSeparator();
+    static ArrayList<Task> taskList = new ArrayList<>(); 
+    public static void main(String[] args) throws IOException {
+        loadTaskFile(taskList);
+        Scanner scanner = new Scanner(System.in);
+        printGreeting();
+        while (true) {
+            String input = scanner.nextLine();
+            String command = input.split(" ", 2)[0]; // Get the first word of the input as the command
+
+            switch (command) {
+                case "bye":
+                    printBye();
+                    scanner.close();
+                    return;
+                case "list":
+                    printList();
+                    break;
+                case "mark":
+                    try {
+                        markList(input);
+                    } catch (InvalidTaskIndexException e){
+                        System.out.println("The task you are trying to edit does not exist!" + nl + line);
+                    }
+                    break;
+                case "unmark":
+                    try {
+                        unmarkList(input);
+                    } catch (InvalidTaskIndexException e){
+                        System.out.println("The task you are trying to edit does not exist!" + nl + line);
+                    }
+                    break;
+                case "todo":
+                    try {
+                        addTodo(input);
+                    } catch (InvalidCommandMessageException e) {
+                        System.out.println("The todo command message is invalid" + nl + line);
+                    }
+                    break;
+                case "deadline":
+                    try {
+                        addDeadline(input);
+                    } catch (InvalidCommandMessageException e) {
+                        System.out.println("The deadline command is invalid" + nl +line);
+                    }
+                    break;
+                case "event":
+                    try {
+                        addEvent(input);
+                    } catch (InvalidCommandMessageException e) {
+                        System.out.println("The event command is invalid" + nl + line);
+                    }
+                    break;
+                case "delete":
+                    try {
+                        deleteTask(input);
+                    } catch (InvalidTaskIndexException | InvalidCommandMessageException e) {
+                        System.out.println("The delete command is invalid" + nl + line);
+                    }
+                    break;
+                default:
+                    try {
+                        throw new InvalidCommandException();
+                    } catch (InvalidCommandException e) {
+                        System.out.println("Please enter a valid command: todo, deadline or event");
+                    }
+            }
+            saveTasks(taskList);
+        }
+    }
```

Comment:
1. This code block can be abstracted to improve the code readability. Usually, methods should not be longer than 30 Lines of Code.
2. There is deep nesting present in this block. You may need to work on abstraction for this as well.

Reference:
https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods
https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting